core/src/main/java/org/apache/iceberg/util/EventParser.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.util;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+import java.util.Map;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.events.CreateSnapshotEvent;
+import org.apache.iceberg.events.IncrementalScanEvent;
+import org.apache.iceberg.events.ScanEvent;
+import org.apache.iceberg.expressions.ExpressionParser;
+
+public class EventParser {
+  private static final String TABLE_NAME = "table-name";
+  private static final String SNAPSHOT_ID = "snapshot-id";
+  private static final String PROJECTION = "projection";
+  private static final String EXPRESSION = "expression";
+  private static final String OPERATION = "operation";
+  private static final String SEQUENCE_NUMBER = "sequence-number";
+  private static final String SUMMARY = "summary";
+  private static final String FROM_SNAPSHOT_ID = "from-snapshot-id";
+  private static final String TO_SNAPSHOT_ID = "to-snapshot-id";
+
+  private EventParser() {
+  }
+
+  public static String toJson(Object event) {
```

Comment:
Question: Would it make sense to add a top level marker such as `Event`, that's maybe even an empty interface or something, so as to add some level of type safety? Is there any drawback to doing that?